Linear regression

thejjserg opened the pull request:
Hello, Team - I found a more elegant method for calculating linear regression using matplot, scipy, and stats.

I've kept my trial-and-error method for calculating linear regression as I'll write a paragraph explaining it, and I'll be comparing the two methods.

GALLP900 commented on us-medical-insurance-costs.ipynb:
```diff
@@ -1365,9 +1367,71 @@
+    "    return [slope, intercept, r, p, std_err]\n",
     "\n",
-    "# Team! We are limited by the constraints"
+    "simple_linear_regression_age = simple_linear_regression(x = age, x_word = \"age\")"
```
Is it possible to extract the name directly from the list instead of specifying it?

GALLP900 commented:
@thejjserg I think the linear regression works amazingly. nice work!! I did a simple test by plotting each variable with itself. this should give a perfect linear fit with R=1, 45deg slope and 0 offset or intercept which works 100% ok. My only comment would be when plotting any x,y variables not only x vs changes. Changes are needed in various lines of the function(s). Would be nice to be able to only specify the x y variables and have the plots to extract the labels directly from the list for example.  
Otherwise, this looks amazing, if @sunilyramjitsingh you dont have more comments we could merge this branch

thejjserg commented:
@GALLP900 great comments! I've added your suggestions and added some changes. I found a function that someone wrote that get the name of a variable and returns it as a string, so that's helpful!

GALLP900 reviewed (approved):
Looks good, thanks for the improvements
